core/results.py

```diff
@@ -375,9 +376,8 @@ def do_write(outfile):
         try:
             do_write(outfile)
         except OSError as e:
-            # If our OSError is because dest is already a directory, we want to handle that. 21 is
-            # the code we get on OS X and Linux, 13 is what we get on Windows.
-            if e.errno in {21, 13}:
+            # If our OSError is because dest is already a directory, we want to handle that.
+            if e.errno in (EISDIR, EACCES):
```

Comment:
I do think maybe leaving the note about how different errors here are due to the different OSes would be helpful in the future. Same in the other file as well.

Reply:
Comments refactored as per suggestion